README.md

```diff
@@ -22,25 +22,24 @@ import (
 )
 
 func main() {
-	logger := logger.NewLogrus(false)
+	log := logger.NewLogrus(false)
 
-	inventory, err := inventory.FromYAMLFile("/go/src/github.com/nornir-automation/gornir/examples/hosts.yaml")
+	file := "/go/src/github.com/nornir-automation/gornir/examples/hosts.yaml"
+	plugin := inventory.FromYAML{HostsFile: file}
+	inv, err := plugin.Create()
 	if err != nil {
-		logger.Fatal(err)
+		log.Fatal(err)
 	}
 
-	gr := &gornir.Gornir{
-		Inventory: inventory,
-		Logger:    logger,
-	}
+	gr := gornir.New().WithInventory(inv).WithLogger(log)
```

Comment:
I am thinking that maybe `New` should require `inv` and `log` as parameters as they are both mandatory.

Reply:
What I thinking is `New` should provide safe defaults instead. Something like an Empty `Inventory` and logging from the standard library `log` package (which doesn't satisfy the `Logger` interface right now). I can address this in a different PR so it's clearly documented.

This way we keep the syntax clean.

Reply:
Ok, I have a different view here, I don't want defaults because I think everything should be a plugin. Also, creating as default an empty inventory has no value as it renders Gornir unusable.

Re using the standard `log` package, it shouldn't be hard adding a plugin that wraps it.

Reply:
Yeah, let's keep it simple for now: `gornir.New()` with no defaults. Let's have a separate discussion on what we want `New` to become.